### PR TITLE
Fix website crashing

### PR DESCRIPTION
**Pull request by trizoza**
Fixes 2 events causing website to crash.

1. If incomplete item is saved Prismic. Fixed by [filtering incomplete items](https://github.com/RichardPK/eh-fm/pull/44/commits/6dd43b3189298594475ece4cc64f0b53139253f3).
2. If there is no current show. Fixed by removing [`currentShow` from `essentialForAllPaths`](https://github.com/RichardPK/eh-fm/pull/44/files#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67R22) condition and adding fallback Components.

**Fallback components**

**Mobile with show**

Screenshot 2022-09-15 at 22 34 14

**Mobile no show**

Screenshot 2022-09-15 at 22 33 57

**Desktop with show**

Screenshot 2022-09-15 at 22 31 26

**Desktop no show**

Screenshot 2022-09-15 at 22 33 14

### CODE REVIEW

File: src/containers/home/Home.js [hunk 1]
```diff
@@ -104,6 +111,7 @@ const HomeContainer = ({ carouselData }) => {
               const sortedData = reverseChronologicalSort(
                 carousel.carousel_items
               );
+              console.log("sortedData", sortedData);
```
Comment (RichardPK) [inline src/containers/home/Home.js]: Probably want to remove this?

File: src/components/current-show/CurrentShow.js [hunk 2]
```diff
@@ -54,60 +54,58 @@ const CurrentShow = ({ currentShow, residentsData }) => {
...
     }
+
+    return "Edinburgh Community Radio.";
   };
 
   return (
     <Wrapper>
       <OnAirWrapper>
-        <OnAir />
+        <OnAir on={Boolean(currentShow)} />
```
Comment (RichardPK) [inline src/components/current-show/CurrentShow.js]: `!!currentShow`?

File: src/components/current-show/CurrentShow.js [hunk 3]
```diff
@@ -54,60 +54,58 @@ const CurrentShow = ({ currentShow, residentsData }) => {
...
-      )}
+      <CurrentShowImageWrapper>{returnImage()}</CurrentShowImageWrapper>
+      <InfoWrapper>
+        {currentShow && prismicShow === SHOW_NOT_FOUND ? (
+          <NameWrapper>
+            <ShowName>{parseShowName(currentShow)}</ShowName>
+          </NameWrapper>
+        ) : null}
+        {currentShow && prismicShow && prismicShow.uid ? (
+          <Link to={`/residents/${(prismicShow && prismicShow.uid) || ""}`}>
```
Comment (RichardPK) [inline src/components/current-show/CurrentShow.js]: Already evaluating `prismicShow && prismicShow.uid` above so fine to assume we have the uid here?

File: src/components/current-show/CurrentShow.js [hunk 4]
```diff
@@ -54,60 +54,58 @@ const CurrentShow = ({ currentShow, residentsData }) => {
...
-          </InfoWrapper>
-        </>
-      )}
+      <CurrentShowImageWrapper>{returnImage()}</CurrentShowImageWrapper>
+      <InfoWrapper>
+        {currentShow && prismicShow === SHOW_NOT_FOUND ? (
+          <NameWrapper>
+            <ShowName>{parseShowName(currentShow)}</ShowName>
+          </NameWrapper>
+        ) : null}
```
Comment (RichardPK) [inline src/components/current-show/CurrentShow.js:79]: My own logic has confused me here ha.

File: src/components/current-show/CurrentShow.js [hunk 5]
```diff
@@ -54,60 +54,58 @@ const CurrentShow = ({ currentShow, residentsData }) => {
...
+                placeholder
+              />
+            </NameWrapper>
+          </Link>
+        ) : null}
+        {!currentShow ? (
+          <NameWrapper>
+            <ShowName>We are currently offline.</ShowName>
+          </NameWrapper>
+        ) : null}
```
Comment (RichardPK) [inline src/components/current-show/CurrentShow.js:104]: Are ternaries necessary for all these? `{ !currentShow &&... ` isn't enough?

File: src/components/nav-bar/logo/Logo.js [hunk 6]
```diff
@@ -9,7 +9,7 @@ const LogoHead = ({ mobile, color }) => {
     <React.Fragment>
       <Wrapper>
         <Link to="/">
-          <StyledLogo color={color} mobile={mobile} />
+          <StyledLogo color={color} mobile={mobile ? "mobile" : "desktop"} />
```
Comment (RichardPK) [inline src/components/nav-bar/logo/Logo.js:12]: at some point it's probably worth changing all of my dumbass boolean prop names to be prefixed with `is...` so `isMobile` to make it clearer what the prop is referring to

File: src/components/players/player/Player.js [hunk 7]
```diff
@@ -12,17 +12,13 @@ import { sanitiseString } from "../../../helpers/PrismicHelper";
...
-    handleVolumeClicked,
-  } = useContext(RadioPlayerContext);
+  const { playing, handlePlayPauseClicked, volume, handleVolumeClicked } =
+    useContext(RadioPlayerContext);
 
   return (
     <Wrapper>
       <OnAirWrapper>
-        <OnAir />
+        <OnAir on={Boolean(currentShow)} />
```
Comment (RichardPK) [inline src/components/players/player/Player.js]: Same as above `!!currentShow`?

File: src/components/players/player/on-air/OnAir.js [hunk 8]
```diff
@@ -4,11 +4,11 @@ import Colors from "../../../../consts/Colors";
 import { Tiny } from "../../../text-elements/index";
 import Devices from "../../../../consts/Devices";
 
-const OnAir = () => {
+const OnAir = ({ on }) => {
```
Comment (RichardPK) [inline src/components/players/player/on-air/OnAir.js]: `isOnAir`? Might as well start the better naming practice now

File: src/components/schedule/Schedule.js [hunk 9]
```diff
@@ -8,26 +8,29 @@ import { getShowInPrismic } from "../../helpers/PrismicHelper";
...
     <Wrapper>
-      <ComingUpText>Coming up...</ComingUpText>
-      <ScheduleItemsWrapper>
-        {scheduleData &&
-          scheduleData.map((scheduleItemData, i) => {
-            const foundShow = getShowInPrismic({
-              residentsData,
-              currentShow: scheduleItemData,
-            });
+      {scheduleData && scheduleData.length ? (
```
Comment (RichardPK) [inline src/components/schedule/Schedule.js]: `scheduleData && !!scheduleData.length &&` might be a away to dodge the ternary?

**Review by RichardPK — APPROVED** [review]
Nice one all just suggestions 👍

File: src/components/current-show/CurrentShow.js (hunk 5 above)
Comment (trizoza) [reply src/components/current-show/CurrentShow.js:104]: There's [a reason not to use](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx) the `&&` for component rendering logic. And it's the few random cases when `!someData` evaluate in 0, and like that you end up with a screen with rogue zero :D

File: src/components/nav-bar/logo/Logo.js (hunk 6 above)
Comment (trizoza) [reply src/components/nav-bar/logo/Logo.js:12]: Hahaha, was thinking that, but it's still quite readable, so might be renaming in future!

File: src/components/schedule/Schedule.js (hunk 9 above)
Comment (trizoza) [reply src/components/schedule/Schedule.js]: Gonna keep ternary for [these reasons](https://github.com/RichardPK/eh-fm/pull/44#discussion_r973617500)

File: src/containers/home/Home.js (hunk 1 above)
Comment (trizoza) [reply src/containers/home/Home.js]: Thanks. Sorry about that. Removed.

File: src/components/current-show/CurrentShow.js (hunk 3 above)
Comment (trizoza) [reply src/components/current-show/CurrentShow.js]: Good spot, [fixed here](https://github.com/RichardPK/eh-fm/pull/44/commits/3df5982b7639ef413d244adf7598904829ba04a4).

File: src/components/players/player/Player.js (hunk 7 above)
Comment (trizoza) [reply src/components/players/player/Player.js]: [Fixed](https://github.com/RichardPK/eh-fm/pull/44/commits/f5dd08b07100643dceef2d2a149e058356c1cbf1)

File: src/components/players/player/on-air/OnAir.js (hunk 8 above)
Comment (trizoza) [reply src/components/players/player/on-air/OnAir.js]: Good point! [Fixed](https://github.com/RichardPK/eh-fm/pull/44/commits/3df5982b7639ef413d244adf7598904829ba04a4)

File: src/components/current-show/CurrentShow.js (hunk 2 above)
Comment (trizoza) [reply src/components/current-show/CurrentShow.js]: [Fixed](https://github.com/RichardPK/eh-fm/pull/44/commits/f5dd08b07100643dceef2d2a149e058356c1cbf1)

**Review by RichardPK — APPROVED** [review]
👍

**Pull request merged**